### PR TITLE
Add contributing options on every page

### PR DESCRIPTION
**Pull request by jseldess**
- `Edit This Page`, which should prompt user to fork the repo
- `Report Doc Issue`, which should open an issue with template text
  and community label
- `Suggest New Content`, which should open an issue with template text
  and community label

To NOT add contributing options to a page, `contribute: false` must be set in the page's front-matter.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1440)

### CODE REVIEW

File: _layouts/page.html [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
 
 <div class="post-header">
+   <div class="contribute dropdown">
+      <button class="btn-contribute dropdown-toggle" type="button" id="dropdownMenu1" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+         <i class="fa fa-github"></i> Contribute
+         <span class="arrow-down"></span>
+      </button>
+      <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
+         <li><a href="https://github.com/cockroachdb/docs/edit/gh-pages{{ page.url | replace: '.html', '.md'}}" target="blank">Edit This Page</a></li>
+         <li><a href="https://github.com/cockroachdb/docs/issues/new?title={{page.title}}%20Doc%20Update%20&body=Re%3A%20%5B{{page.title}}%5D(https%3A%2F%2Fcockroachlabs.com/docs/{{page.url}})%0A%0APriority%20(Low‚%20Medium‚%20High)%3A%0A%0A%23%23%20Issue%20Description%3A%0A%0A%23%23%20Suggested%20Resolution%20&labels=&labels=community" target="blank">Report Doc Issue</a></li>
```
Comment (benesch) [inline _layouts/page.html]: I think you can kill the empty `&label=`?

File: _layouts/page.html [hunk 2]
```diff
@@ -3,6 +3,17 @@
...
+         <li><a href="https://github.com/cockroachdb/docs/issues/new?title=New%20Doc%20Proposal%20&body=Priority%20(Low‚%20Medium‚%20High)%3A%0A%0A%23%23%20Title%0A%0A%0A%23%23%20Description%0A%0A%0A%23%23%20Outline%0A%0A%0A%23%23%20Expected%20Audience&labels=community" target="blank">Suggest New Content</a></li>
```
Comment (benesch) [inline _layouts/page.html]: To be valid HTML, these `&`s (here and the line above) should be escaped as `&amp;`, though most browsers don't really care.

File: _layouts/page.html (hunk 1 above)
Comment (jseldess) [inline _layouts/page.html]: Good catch. Done.

File: _layouts/page.html (hunk 2 above)
Comment (jseldess) [inline _layouts/page.html]: Thanks, @benesch. I think I caught them all now.

**Comment by jseldess** [comment]
TFTR, @benesch!

@sploiselle, @mjibson, @spencerkimball, @petermattis, @bdarnell, if you have additional thoughts/feedback, please share. 

@kuanluo, can you chime in on styling? I've tried to use some other "button" conventions here, while trying to keep these options a bit less in-your-face than a real button. But the drop-down coloring should still be styled. Interested in your overall design guidance, but some specific questions:
- Is the placement good? 
- What should happen when the page title is long? Should the page title wrap early due to the contributing options (currently what happens)? Or should the contributing options be repositioned below the title?
- How should these contributing options behave on smaller screens? It's not likely that someone will "Edit This Page" on a phone, but they may want to report and issue or suggest a new topic...

**Comment by jseldess** [comment]
For clarity, this is what it looks like: 

screen shot 2017-05-19 at 10 54 48 am

**Comment by bdarnell** [comment]
LGTM

**Comment by kuanluo** [comment]
Great idea, @jseldess! Some ideas to your questions:

> Is the placement good?

I like the current placement, and we can certainly track the usage and improve from there. One idea is to repeat the button at the end of the page, next to "Was this page helpful" to capture those who read the whole way through and might have ideas/suggestions.

> What should happen when the page title is long? Should the page title wrap early due to the contributing options (currently what happens)? Or should the contributing options be repositioned below the title?

I prefer having the contribution options be repositioned below the title. My hypothesis is that the second option would require less work and be equally effective.

> How should these contributing options behave on smaller screens? It's not likely that someone will "Edit This Page" on a phone, but they may want to report and issue or suggest a new topic...

If we position the contributing option below the title, per the question above, then on mobile, it should behave the same as on desktop. Currently, it looks like the dropdown is slightly cut off on the right edge in the smaller window, but other than that, it works just fine. Another suggestion would be to place "edit this page" as the last option in the dropdown, so report and suggest are prioritized?

On the design of the dropdown itself, I overall love it. Just a few nits to make sure we don't have different type spaces floating around on the site. :)
1. The title contribute seems to have a heavier type weight than the site navigations above. Can you make sure the all caps stylings are the consistent?
1. Would be great to have the dropdown options follow the same type/line-height as the docs nav on the left. The current treatment looks a bit light and tight. The hover would be a text color change to green.
2. For the dropdown container, we don't use borders (unless it's a screenshot) and the drop shadow is a bit softer. Check out the "Run on any cloud or on-premise" module for the implementation details. 

Let me know if you have any questions!

**Comment by jseldess** [comment]
@kuanluo, thanks for the feedback. I still need to deal with positioning, but please take another look at the styling of the menu. I've tried to address 2 & 3 above. 

As for 1, I can't find relevant differences to the topnav. Weight and all caps seems correct. But please let me know if I'm missing something. 

screen shot 2017-05-21 at 11 17 02 pm

**Comment by jseldess** [comment]
@kuanluo, figured out how to make `CONTRIBUTING` match the topnav styling. Still need to work on positioning on smaller screens though. 

screen shot 2017-05-22 at 10 28 44 am

**Comment by kuanluo** [comment]
LGTM, @jseldess! Thank you for taking the time to fix the styling. Much appreciated!

**Comment by sploiselle** [comment]
So cool, @jseldess! LGTM. Have one really minor nit.

---

Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[_layouts/page.html, line 14 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1440#-KkW6KfxbkrN7IRKyihx-r1-14:-KkkinH3dbp4awTvhD6E:bj8wkdy) ([raw file](https://github.com/cockroachdb/docs/blob/342f7cae04e3498a6201a04c1b669e6a2c9a48f1/_layouts/page.html#L14)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Thanks, @benesch. I think I caught them all now.
</blockquote></details>

Same thing here with the line feed 

---

*[_layouts/page.html, line 14 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1440#-KkkiAutPTy-Vbq1Y__7:-KkkiAutPTy-Vbq1Y__8:b-lxa6oh) ([raw file](https://github.com/cockroachdb/docs/blob/5a9074dcedd42780f73f2ad7ee0473899d8dad0a/_layouts/page.html#L14)):*
> ```HTML
>       <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
>          <li><a href="https://github.com/cockroachdb/docs/edit/gh-pages{{ page.url | replace: '.html', '.md'}}" target="blank">Edit This Page</a></li>
>          <li><a href="https://github.com/cockroachdb/docs/issues/new?title={{page.title}}%20Doc%20Update%20&amp;body=Re%3A%20%5B{{page.title}}%5D(https%3A%2F%2Fcockroachlabs.com/docs/{{page.url}})%0A%0APriority%20(Low‚%20Medium‚%20High)%3A%0A%0A%23%23%20Issue%20Description%3A%0A%0A%23%23%20Suggested%20Resolution%20&amp;labels=community" target="blank">Report Doc Issue</a></li>
> ```

I'd add a line feed (`%0A`) to the last item so there's a pre-populated space users can begin typing.

`<li><a href="https://github.com/cockroachdb/docs/issues/new?title={{page.title}}%20Doc%20Update%20&amp;body=Re%3A%20%5B{{page.title}}%5D(https%3A%2F%2Fcockroachlabs.com/docs/{{page.url}})%0A%0APriority%20(Low‚%20Medium‚%20High)%3A%0A%0A%23%23%20Issue%20Description%3A%0A%0A%23%23%20Suggested%20Resolution%20%0A&amp;labels=community" target="blank">Report Doc Issue</a></li>`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1440#-:-Kkki11NNHhOK2YXG4CX:b-rqpzs3)*

**Comment by jseldess** [comment]
*[_layouts/page.html, line 14 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1440#-KkkiAutPTy-Vbq1Y__7:-KkkoHN7PZjOBnjMRiMI:bu8u455) ([raw file](https://github.com/cockroachdb/docs/blob/5a9074dcedd42780f73f2ad7ee0473899d8dad0a/_layouts/page.html#L14)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary><blockquote>

I'd add a line feed (`%0A`) to the last item so there's a pre-populated space users can begin typing.

`<li><a href="https://github.com/cockroachdb/docs/issues/new?title={{page.title}}%20Doc%20Update%20&amp;body=Re%3A%20%5B{{page.title}}%5D(https%3A%2F%2Fcockroachlabs.com/docs/{{page.url}})%0A%0APriority%20(Low‚%20Medium‚%20High)%3A%0A%0A%23%23%20Issue%20Description%3A%0A%0A%23%23%20Suggested%20Resolution%20%0A&amp;labels=community" target="blank">Report Doc Issue</a></li>`
</blockquote></details>

Done. Thanks, Sean.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1440)*

**Comment by jseldess** [comment]
*[_layouts/page.html, line 14 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1440#-KkW6KfxbkrN7IRKyihx-r1-14:-KkkoO4BYId9wUZTrwqk:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/342f7cae04e3498a6201a04c1b669e6a2c9a48f1/_layouts/page.html#L14)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary><blockquote>

Same thing here with the line feed 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1440)*

**Comment by benesch** [comment]
How's this, @jseldess? 

screenshot 2017-05-22 20 14 54

I force-pushed your branch, so you'll want to `git fetch origin && git checkout contribute-link && git reset --hard origin/contribute-link` to get my changes locally, should you need them. Fair warning: that command will blow away any local changes you have on that branch.

**Comment by jseldess** [comment]
Perfect, @benesch! Thanks very much.

**Comment by benesch** [comment]
My pleasure, @jseldess!

**Comment by sploiselle** [comment]
@benesch There's nothing you can't do! Can I get a tl;dr of what you did to the `<h1>` and the Contribute button?

**Comment by benesch** [comment]
Yeah, totally! Are you familiar with [flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)? It's a new layout model for CSS that makes it possible or at least a whole lot less painful to achieve certain layouts. This contributing button is a good example of where flexbox shines; without flexbox, I think you'd need JavaScript.

Basically, you opt-in to flexbox on a per-element basis by specifying `display: flex`. I did this for the parent container, `.post-header`, of the page title and contributing button. By default, flexbox won't let elements wrap, so I opted in with `flex-wrap: wrap`. Then, the `justify-content: space-between` asks that any extra space on the line be placed between the child elements. So when the page title and contributing button are on the same line, the extra space is put between them, pushing the contributing button all the way to the right. When the contributing button is wrapped to the next line, there's only one element per line, so they both appear at the start of their lines.

**Comment by benesch** [comment]
Oh, I forgot to mention: all that margin hacking (-30px on the container, 30px on the children) ensures that the title and the contribute button always have 30px of space *from each other*, but no space from the edge of the container. Adding a `margin-left: 30px` on the contribute button doesn't quite work, because it'll be indented 30px when it wraps. Adding a `margin-right: 30px` on the page title also doesn't work, because it means it can't use the last 30px of the line to display a word it otherwise could fit.

**Pull request merged**